experiments/process_sandbox/library_runner.cc

```diff
@@ -335,73 +485,123 @@ namespace sandbox
       }
     }
   };
+
 }
 
 char* ExportedLibrary::type_encoding(int idx)
 {
   return functions.at(idx)->type_encoding();
 }
 
-int main(int, char** argv)
+namespace
 {
+  SNMALLOC_SLOW_PATH
+  void bootstrap()
+  {
 #ifdef USE_CAPSICUM
-  cap_enter();
+    cap_enter();
 #endif
-  void* addr = (void*)strtoull(argv[1], nullptr, 0);
-  size_t length = strtoull(argv[2], nullptr, 0);
-  // fprintf(stderr, "Child starting\n");
-  // printf(
-  //"Child trying to map fd %d at addr %p (0x%zx)\n", SharedMemRegion, addr,
-  // length);
-  void* ptr = mmap(
-    addr,
-    length,
-    PROT_READ | PROT_WRITE,
-    MAP_FIXED | MAP_ALIGNED(35) | MAP_SHARED | MAP_NOCORE,
-    SharedMemRegion,
-    0);
-  // printf("%p\n", ptr);
-  if (ptr == MAP_FAILED)
-  {
-    err(1, "Mapping shared heap failed");
+    void* addr = nullptr;
+    size_t length = 0;
+    // Find the correct environment variables.  Note that libc is not fully
+    // initialised when this is called and so we have to be very careful about
+    // the libc function that we call.  We use the `environ` variable directly,
+    // rather than `getenv`, which may allocate memory.
+    for (char** e = environ; *e != nullptr; e++)
+    {
+      char* ev = *e;
+      const char ev_name[] = "SANDBOX_LOCATION=";
```

Comment:
Who sets this? Shouldn't the error message be a little more descriptive if the user is expected to know this?

I know this is an experiment, but without reading the github description of the PR, this would have been lost to me.

We can't expect people to git-blame to find the PR number and read the thread.